plugin/src/main/java/eu/maveniverse/maven/njord/plugin3/CheckArtifactsAvailabilityMojo.java

```diff
@@ -0,0 +1,255 @@
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.impl.RepositoryConnectorProvider;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RepositoryPolicy;
+import org.eclipse.aether.spi.connector.ArtifactDownload;
+import org.eclipse.aether.spi.connector.RepositoryConnector;
+import org.eclipse.aether.transfer.NoRepositoryConnectorException;
+
+/**
+ * A mojo that checks availability for all artifacts on given remote repository. The mojo by default blocks/waits
+ * for configured time and polls remote repository at given intervals, but can be used for "one pass" checks as well.
+ * <p>
+ * By default, the mojo will use {@link ArtifactStore} as "artifacts source" and
+ * {@link ArtifactStorePublisher#targetReleaseRepository()} or {@link ArtifactStorePublisher#targetSnapshotRepository()}
+ * as source of remote repository, but user can configure list of artifacts and remote repository directly as well.
+ * <p>
+ * This Mojo is useful in complex workflows/scenarios, where something (fx a build) should happen after publishing done.
+ * While some services do provide status of (non-atomic) publishing, this mojo checks the "real thing": when it
+ * succeeds, the artifacts can be resolved 100% by builds from given remote repository. Hence, this mojo
+ * can work with all publishers available out there, and even with in-house MRMs solutions as well.
+ */
+@Mojo(name = "check-artifacts-availability", threadSafe = true, requiresProject = false, aggregator = true)
+public class CheckArtifactsAvailabilityMojo extends PublisherSupportMojo {
+    /**
+     * If using {@link ArtifactStore} as artifact source, whether source store should be dropped after successful operation.
+     */
+    @Parameter(required = true, property = "drop", defaultValue = "true")
+    private boolean drop;
+
+    /**
+     * Should the mojo block/wait for artifacts to become available, or should just perform one-pass of check.
+     */
+    @Parameter(required = true, property = "wait", defaultValue = "true")
+    private boolean wait;
+
+    /**
+     * If mojo set to {@link #wait}, the total allowed wait duration (as {@link Duration} string).
+     */
+    @Parameter(required = true, property = "waitTimeout", defaultValue = "PT1H")
+    private String waitTimeout;
```

Comment:
One thing, I would add is also a delay before starting to check.

For Quarkus, for instance, we know we have no chance of having them before 20 minutes. Now that's something I could add in the script on our side but I think it would be good practice to not encourage for not testing right after publication and having it documented here with a parameter would help.

Also even for the easy case, I would have a couple of minutes delay by default.

Reply:
Agreed, will add `waitDelay` that is duration that must pass BEFORE first check is done.